src/Avalonia.Controls/Grid.cs

```diff
@@ -161,44 +168,50 @@ public bool ShowGridLines
         }
 
         /// <summary>
-        /// Returns a ColumnDefinitions of column definitions.
+        /// A list of <see cref="ColumnDefinition"/> objects defined on this instance of <see cref="Grid"/>.
         /// </summary>
         public ColumnDefinitions ColumnDefinitions
         {
             get
             {
-                if (_data == null) { _data = new ExtendedData(); }
-                if (_data.ColumnDefinitions == null) { _data.ColumnDefinitions = new ColumnDefinitions() { Parent = this }; }
+                if (_data is null)
```

Comment:
Putting initialization in the getter for a styled property isn't a good idea because it won't be run when calling e.g. `GetValue` or `GetObservable`, for example this fails:

```
        [Fact]
        public void ColumnDefinitions_Should_Not_Be_Null()
        {
            var grid = new Grid();

            Assert.NotNull(grid.GetValue(Grid.ColumnDefinitionsProperty));
        }
```

It's not really possible to have lazily-initialized styled properties; you should initialize the properties in the constructor.

Reply:
@grokys I don't remember about WPF, but UWP has default value factory in the DependencyProperty
See https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.propertymetadata.createdefaultvaluecallback?view=winrt-19041

It also helps in cases, when default value is mutable - https://devblogs.microsoft.com/oldnewthing/20191002-00/?p=102950

Reply:
Although it doesn't provide access to dependency object itself as I remember